src/transports/tcp/TcpConnectionAcceptor.cpp

```diff
@@ -114,4 +114,12 @@ void TcpConnectionAcceptor::stop() {
       [this] { serverSocket_.reset(); });
   serverThread_.reset();
 }
+
+folly::Optional<uint16_t> TcpConnectionAcceptor::listeningPort() const {
+  if (!serverSocket_) {
+    return folly::none;
```

Comment:
That's more confusing than it needs to be.  The point of the function is to return the port currently being listened on.  If there is no port being listened on, then we should return some kind of "null" value.  Returning the value in options.port is odd because that's the port we _wanted_ to listen on, not the actual port we _are_ listening on (e.g. wanted to listen on TCP port 0, actually listening on TCP port 25023).

tl;dr I don't think this function should return port N if the acceptor isn't actually listening on port N at the given moment.